orchestration/dagster_orchestration/hca_manage/check.py

```diff
@@ -38,12 +48,251 @@ def check_data(args: argparse.Namespace, host: str, parser: argparse.ArgumentPar
     else:
         project = args.project
 
-    hca = HcaManage(environment=args.env,
-                    project=project,
-                    dataset=args.dataset,
-                    data_repo_client=get_api_client(host=host))
+    hca = CheckManager(environment=args.env,
+                       project=project,
+                       dataset=args.dataset,
+                       data_repo_client=get_api_client(host=host))
 
     if remove:
         hca.remove_all()
     else:
         hca.check_for_all()
+
+
+@dataclass
+class CheckManager(SoftDeleteManager):
+
+    @cached_property
```

Comment:
I don't think this needs to be cached, it's used exactly once.

Reply:
So, does this actually need to be cached? I'm curious why this decorator is being applied here.

Reply:
confused why this has been marked as resolved - drew's right, i'm unclear on why we want to cache this (and if we do, we should use the python core version of `cached_property`)